README.md

```diff
@@ -0,0 +1,32 @@
+# ERC2048
+
+Inspired by [ERC404](https://github.com/Pandora-Labs-Org/erc404), ERC2048 is an experimental, mixed ERC20 / ERC721 implementation with native liquidity and fractionalization.
+
+In ERC404, there is a 1:1 mapping between fungible tokens (FT) and non-fungible tokens (NFT). This caused two challenging issues when using the ERC404 protocol:
+
+1. When transferring a large amount of tokens, the times of minting and burning NFTs equal to the token amount, it'll cost too much Gas. In the initial version of ERC404 code, it try to use whitelist to make it possible to transfer large amount, but it proved ineffective and caused a bug.
+2. With the 1:1 mapping between NFTs and tokens, different rarities of NFTs lose their value. Even if a green Pandora box has a lower probability of generation compared to a red Pandora box, they are both equal to 1 token, right?
+
+Inspired by the [2048 game](https://en.wikipedia.org/wiki/2048_(video_game)), in the ERC2048 protocol, we introduce the concept of "Level" for NFTs. Similar to the game, when you collect two NFTs of the same Level, they automatically upgrade to a higher Level NFT! Let's provide some examples:
+
+1. Alice purchases a Level(0) NFT, which also means she receives 1 FT.
+2. Alice buys a token, which also means she acquires a Level(0) NFT. Since she already owns a Level(0) NFT, these two Level(0) NFTs will merge into a Level(1) NFT.
```

Comment:
2. Alice buys a FT, which also means she acquires a Level(0) NFT. Since she already owned a Level(0) NFT, these two Level(0) NFTs will merge into a Level(1) NFT.